e2e-webdriver/src/app.e2e.ts

```diff
@@ -1,28 +1,28 @@
-import { AppPage } from './page-objects/app.po';
+import { ProductMasterPage } from './page-objects/product-master.po';
 import { ProductDetailPage } from './page-objects/product-detail.po';
 
 describe('workspace-project App', async () => {
-    let page: AppPage;
+    let page: ProductMasterPage;
     let detailPage: ProductDetailPage;
 
     beforeEach(() => {
-        page = new AppPage();
+        page = new ProductMasterPage();
```

Comment:
Die Idee bei den Page Objects laut WDIO ist, dass man diese Klassen nicht hier zu initialisieren braucht, sondern dass es eine Instanz dieser Klasse gibt, sobald sie importiert wird.

Also:

```ts
// page object
class ProductMasterPage {
}
export const productMasterPage = new ProductMasterPage();

// spec
import { productMasterPage } from 'product-master.po';

await expectAsync(productMasterPage.title).toHaveText('foo');
```

Reply:
Ok, ist erledigt. Meinst du das macht auch Sinn für Playwright? Da dort nun die globale Instanz von 'page' genutzt wird und die page damit nicht mehr im Konstruktor der Page Objects übergeben wird (siehe [hier](https://github.com/flntzr/angular-redux-shop/pull/4/commits/694e45d1a8f0273d3077f12218cd92b38c7858f4)), könnte man das ähnlich umsetzen wie in Webdriver.
Der Standard scheint dort aber zu sein, immer eine neue Instanz der POs zu erstellen: https://github.com/microsoft/playwright/issues/1604#issuecomment-655432819 oder https://playwright.dev/docs/pom/

Reply:
Ja, gute Frage. Da es in der Doku und Community nicht so gemacht wird, würde ich mich daran halten, wie es dort ist.
Was wir aber wahrscheinlich nicht machen müssen, ist, die Page Objects mit jedem Test neu zu erzeugen (`beforeEach`). Es würde denk ich reichen, sie einmalig zu erzeugen.